core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1234,8 +1234,9 @@ class ReplicaManager(val config: KafkaConfig,
                                fetchOffset: Long,
                                currentTimeMs: Long): Option[Int] = {
     partition.leaderReplicaIdOpt.flatMap { leaderReplicaId =>
```

Comment:
I think that this line implies the leader, no?

Reply:
@dajac In fact, when the local replica is a follower or changes from a Leader to a Follower, `partition.leaderReplicaIdOpt.flatMap { leaderReplicaId => `refers to the latest leader node. But it does not confirm whether it is the leader itself.
This will result in a case, if the local broker  is follower replica, the follower will also perform the preferred read replica selection. But actually we just want the leader to execute the process.
I implemented a unit test. Confirmed that follower will execute preferred read leader selection.

Reply:
Looks like I assumed that leaderReplicaIdOpt would only be set if the broker was the leader. We should check if the docs need to be clarified (and close the test coverage gap, of course).

Reply:
We should add a comment to `leaderReplicaIdOpt` in the `Partition` class to explain it a bit better. Could you do it please?

Reply:
a comment has added to leaderReplicaIdOpt in the Partition.